src/core/syscalls/deprecated_business_logic_syscall_handler.rs

```diff
@@ -506,38 +513,235 @@ where
         Ok(tx_info_ptr_temp)
     }
 
-    fn library_call(
+    pub(crate) fn storage_read(
+        &mut self,
+        vm: &mut VirtualMachine,
+        syscall_ptr: Relocatable,
+    ) -> Result<(), SyscallHandlerError> {
+        let request = if let DeprecatedSyscallRequest::StorageRead(request) =
+            self.read_and_validate_syscall_request("storage_read", vm, syscall_ptr)?
+        {
+            request
+        } else {
+            return Err(SyscallHandlerError::ExpectedGetBlockTimestampRequest);
+        };
```

Comment:
This will look simpler with `let StorageRead(request) = self.read... else { return };` syntax.